### PR TITLE
openiotsdk: Enable build with GCC 12

### PR DESCRIPTION
**Pull request by pan-**
TF-M fails to link with GCC 11.3 and onward as new warning are raised at link time and the linking step disallow warning. 
To workaround the issue we disable link warnings for TF-M by patching it.

This PR is required by https://github.com/project-chip/connectedhomeip/pull/26133 .

### CODE REVIEW

**Review by ATmobica — APPROVED** [review]

**Review by tehampson — APPROVED** [review]

File: config/openiotsdk/cmake/sdk.cmake [hunk 1]
```diff
@@ -42,6 +42,22 @@ FetchContent_Declare(
...
+    GIT_TAG         d0c0a67f1b412e89d09b0987091c12998c4e4660
+    GIT_SHALLOW     OFF
+    GIT_PROGRESS    ON
+    # Note: This prevents FetchContent_MakeAvailable() from calling
+    # add_subdirectory() on the fetched repository. TF-M needs a
+    # standalone build because it relies on functions defined in its
+    # own toolchain files and contains paths that reference the
+    # top-level project instead of its own project.
+    SOURCE_SUBDIR   NONE
+    PATCH_COMMAND   git reset --hard --quiet && git clean --force -dx --quiet && git apply ${CMAKE_CURRENT_LIST_DIR}/tf-m.patch
```
[the remark this answers is not in the document]
Comment (pan-) [reply config/openiotsdk/cmake/sdk.cmake:58]: This `FetchContent` directives fetch as specific version of TF-M. TF-M is built as a separate binary from the main application. Matter users that use this port should not be impacted by additional issues. 

The next step for me is to get this actually fixed in TrustedFirmware M. I engaged with the team about this issue and they had it already tracked. Once it is updated I will update this part of the Open IoT SDK port. 

I can had an extra comment explaining the impact of the patch though.

[the remark this answers is not in the document]
Comment (tehampson) [reply config/openiotsdk/cmake/sdk.cmake:58]: SGTM

[the remark this answers is not in the document]
Comment (tehampson) [reply config/openiotsdk/cmake/sdk.cmake:58]: >Matter users that use this port should not be impacted by additional issues.

I was just trying to say that while we suppress `-fatal-warnings` it is possible that someone accidentally adds some issue in. If the plan is to remove this soon after it is fixed in TrustedFirmware M. I was simply suggesting adding it in with a possible github issue for tracking sake that it doesn't get forgotten about

**Review by bzbarsky-apple — APPROVED** [review]

File: config/openiotsdk/cmake/sdk.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (pan-) [reply config/openiotsdk/cmake/sdk.cmake:58]: I created an issue to remove the workaround: https://github.com/project-chip/connectedhomeip/issues/26176

**Pull request merged**